src/App.tsx

```diff
@@ -1,61 +1,172 @@
+import { FormEvent, useState } from 'react';
 import './App.scss';
+import {
+  FormControl, InputLabel, MenuItem, Select,
+  Button,
+  TextField,
+  Container,
+  Box,
+} from '@mui/material';
+import usersFromServer from './api/users';
+import { TodoList } from './components/TodoList';
+import { getTodosInfo } from './helpers/getTodosInfo';
+import { removeByPattern } from './helpers/removeByPattern';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const todosInfo = getTodosInfo();
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState<string>('');
+  const [todos, setTodos] = useState(todosInfo);
+  const [errorTitle, setErrorTitle] = useState('');
+  const [errorSelectedUser, setErrorSelectedUser] = useState('');
+
+  const changeUser = (id: string) => {
+    if (id !== '') {
+      setErrorSelectedUser('');
+    }
+
+    setSelectedUser(id);
+  };
+
+  const changeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newTitle = event.target.value;
+
+    if (newTitle !== '') {
+      setErrorTitle('');
+    }
+
+    const pattern = /[@^#$/%^&*()_+-\\{}[\]]/g;
+    const patternedTitle = removeByPattern(newTitle, pattern);
+
+    setTitle(patternedTitle);
+  };
+
+  const clearForm = () => {
+    setTitle('');
+    setSelectedUser('');
+  };
+
+  const handleSubmit = (event:FormEvent) => {
+    event.preventDefault();
+
+    const trimedTitle = title.trim();
+
+    if (!trimedTitle || !selectedUser) {
+      setErrorTitle(trimedTitle ? '' : 'Please enter a title');
+      setErrorSelectedUser(selectedUser ? '' : 'Please choose a user');
+
+      return;
+    }
+
+    const userInfo = usersFromServer
+      .find(user => user.id === +selectedUser);
+
+    if (!userInfo) {
+      throw new Error('User not found');
+    }
+
+    setTodos((prev) => {
```

Comment:
Use more descriptive name.
```suggestion
    setTodos((prevTodos) => {
```